From f060be5db755f4cf38ac6556913a942ea21f57cc Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Jun 2025 18:23:15 +0300
Subject: [PATCH] [JSW-670] jgs: replace rst arc with pm

In JGS, reset arc on compute is replaced with
pm arc. Hence, this patch takes care of those
changes.

Change-Id: I4350f49623e413dd7b75f59255811bc93062ef1f
---

diff --git a/arch/jgs/arc/palad.c b/arch/jgs/arc/palad.c
index 69ee881..9c420a1 100644
--- a/arch/jgs/arc/palad.c
+++ b/arch/jgs/arc/palad.c
@@ -359,7 +359,7 @@
 #else
 void btldr_parc_prepare(void)
 {
-	regmap_writel(0, mmCOMPUTE_0_CFG_PSOC_ARC_RST_ARC_BASE + mmARC_ARC_RST);
+	regmap_writel(0, mmCOMPUTE_0_CFG_PSOC_ARC_PM_ARC_BASE + mmARC_ARC_RST);
 }
 #endif
 
@@ -416,7 +416,7 @@
 	if (!val)
 		return;
 
-	hw_mem_copy(COMP0_RST_ICCM_BASE, PREBOOT_TEXT_BASE_FULL + COMP_PREBOOT_MAX_SZ,
+	hw_mem_copy(COMP0_PM_ICCM_BASE, PREBOOT_TEXT_BASE_FULL + COMP_PREBOOT_MAX_SZ,
 		COMP_RST_ICCM_SIZE, false, false);
 }
 #endif
